src/firefly/js/charts/ChartDataType.js

```diff
@@ -0,0 +1,50 @@
+/*
+ * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
+ */
+
+import {DATATYPE_XYCOLS} from './ChartDataTypeXYCols.js';
+import {DATATYPE_HISTOGRAM} from './ChartDataTypeHistogram.js';
+import {logError} from '../util/WebUtil.js';
+
+/**
+ * @global
+ * @public
+ * @typedef {Object} ChartDataType - an object which specifies how to get chart data
+ * @prop {string} id - unique chart data type id
+ * @prop {Function} fetchData - function to load chart data element data: fetchData(dispatch, chartId, chartDataElementId)
+ * @prop {Function} fetchParamsChanged - function to determine if fetch is necessary: fetchParamsChanged(oldOptions, newOptions)
+ * @prop {Function} getUpdatedOptions - function to resolve the options, which depend on table or chart data getUpdatedOptions(xyPlotParams, tblId, data)
+ */
+
+const chartDataTypes = [];
+
```

Comment:
I am struggle with this code. I am really torn.  It is effectively keeping application state outside of the store. I solved this problem with drawing layer by defining them in ReduxFlux.js (line 110) and them creating the reducing function with the factory (line 129).  However, it does make things more complex.  Might be worth Tatiana, Loi, and Trey talking about tomorrow.